### ISSUE TITLE
o2: connector not working

### ISSUE THREAD

**Issue by GoogleCodeExporter** [satyajit319/websmsdroid#333]
```
as i got tons of mails but nobody writing down a bug report i do this here:

o2 seems to be broken.

i've been told by some users, that o2's app isn't working either.
some told me that the connector works some time.

when using the website to check if it's working, it incredible slow. i assume 
the connector is just running into a timeout.

please make sure, tweaks are disabled.
```

Original issue reported on code.google.com by `felix.bechstein` on 16 Sep 2010 at 4:15

**Comment by GoogleCodeExporter** [comment]
```
some are getting http 500 errors,
some are getting messages login failed.

*all this seems to be not a bug of websms!*
```

Original comment by `felix.bechstein` on 16 Sep 2010 at 4:17

**Comment by GoogleCodeExporter** [comment]
```
I have the problem since 2 days, that the error message appears: "Username or 
Password wrong" (Benutzername oder Passwort falsch)." After rebooting the 
Device, everythink is working for some hours.
```

Original comment by `zink.joc...@googlemail.com` on 17 Sep 2010 at 12:02

**Comment by GoogleCodeExporter** [comment]
```
Don't know if its the same issue but i sometimes get a captcha to fill out - 
which i never got before :P
Also if I click "send" BEFORE I see how many WebSMS at o2 are left, the message 
won't get sent. The notification icon won't show up in the Statusbar.
But when i see the number of left SMS, it seems to always work.
```

Original comment by `gech...@gmail.com` on 17 Sep 2010 at 12:46

**Comment by GoogleCodeExporter** [comment]
```
all this is matching to the very slow website.
the connector does not have a fast api.
it has to load all the webpage several times.

if you run it in parallel, it is even worse.
cleaning up memory with rebooting speeds up the process a bit..
```

Original comment by `felix.bechstein` on 17 Sep 2010 at 5:08

**Comment by GoogleCodeExporter** [comment]
```
For me (HTC Desire/Froyo) it is working exactly once after reboot.
By any second attempt, even if only updating the number of left WebSMS, i get a 
"wrong username and password"-error.
```

Original comment by `kboss...@gmail.com` on 18 Sep 2010 at 2:31

**Comment by GoogleCodeExporter** [comment]
```
please check this apk.
i disabled manually setting the background service to foreground. may be the 
stystem has issues with that.
```

Original comment by `felix.bechstein` on 18 Sep 2010 at 11:45
- Changed state: **Started**

Attachments:
- [ConnectorO2-release.apk](https://storage.googleapis.com/google-code-attachments/websmsdroid/issue-333/comment-6/ConnectorO2-release.apk)

**Comment by GoogleCodeExporter** [comment]
```
anx comments?
```

Original comment by `felix.bechstein` on 19 Sep 2010 at 1:56

**Comment by GoogleCodeExporter** [comment]
```
I had the same problem with wrong username password right after the last WebSMS 
update.
Tried to enter a bug report, but this was too complicated on my Galaxy S.
Today I Googled from the laptop, if anyone else has similar probs, and if there 
is a solution - and yes and yes.
After installing the ConnectorO2-release.apk, posted in this forum, the number 
of left WebSMS ans sending SMS is working again!
Thank you.


```

Original comment by `dilberti...@googlemail.com` on 19 Sep 2010 at 3:03

**Comment by GoogleCodeExporter** [comment]
```
can someone confirm this fix?
```

Original comment by `felix.bechstein` on 19 Sep 2010 at 8:31
- Changed state: **ShouldBeFixed**

**Comment by GoogleCodeExporter** [comment]
```
Confirm: Fix seems to work!

Had this issue for two days now. Always got the error message "o2: Falscher 
Benutzernmane oder Passwort angegeben!" when trying to update the number of 
free SMS or sending a SMS. Now it seems to work again (tried both - updating 
and sending to my own cell phone).

Since i was able to send one lonely SMS the last two days this might be a 
coincidence but I don't think so. :-)

Thanks for the update!
```

Original comment by `lars.roh...@gmail.com` on 20 Sep 2010 at 7:25

**Comment by GoogleCodeExporter** [comment]
```
It seems that it send one sms today without problems. But trying to send 
another one results in: "o2: Falscher Benutzernmane oder Passwort angegeben!"
Even updating the counter ends up with that message.
```

Original comment by `gech...@gmail.com` on 20 Sep 2010 at 12:28

**Comment by GoogleCodeExporter** [comment]
```
Fix worked for two SMS and the status check, but now it failed again...
Using the O2 WebSMS Android app at the same time worked without probs, so I 
doubt, that it is related to a too slow O2 communicator as only reason...
```

Original comment by `dilberti...@googlemail.com` on 20 Sep 2010 at 8:53

**Comment by GoogleCodeExporter** [comment]
```
please try the previous version of the connector:
http://code.google.com/p/websmsdroid/downloads/detail?name=Connector-o2-1.3.3.ap
k
```

Original comment by `felix.bechstein` on 21 Sep 2010 at 4:45
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
After one day without any problems (sent 2 SMS and made a lot of update checks) 
I got the error message again. (Didn't try the previous version though.)
```

Original comment by `lars.roh...@gmail.com` on 21 Sep 2010 at 7:03

**Comment by GoogleCodeExporter** [comment]
```
Just send 4 SMS in a row with the previous version (1.3.3) and i got them 
answered - so it works in the old version :P
```

Original comment by `gech...@gmail.com` on 21 Sep 2010 at 12:34

**Comment by GoogleCodeExporter** [comment]
```
Today it doesnt work any longer with the new fix
```

Original comment by `chiefte...@gmail.com` on 22 Sep 2010 at 3:59

**Comment by GoogleCodeExporter** [comment]
```
and what's with the previous version?
does it fix the problem in long term for all of you?
```

Original comment by `felix.bechstein` on 23 Sep 2010 at 5:12
- Changed state: **NeedMoreInfos**

**Comment by GoogleCodeExporter** [comment]
```
Well, I try to give some information, too. Hope it will help a little.
When I try to send a message an error occurs which sais "o2: Falscher 
Benutzernmane oder Passwort angegeben!". I cannot fix it by restarting the 
device or resetting the connector settings or anything. I just can fix the 
problem by de- and reinstalling the O2-Connector and then setting the connector 
settings again.
After several hours of not using websms the problem shows up again. I 
deactivated the options "Autoupdate" and "Autoexit" to see whether this is the 
source of the problem...but it is not. I also deactivated Tweak Connector but 
that doesn´t change anything with the problem.

What I am gonna do now is to try the previous version, but I guess that won´t 
help, cause if I remember right the error occured the first time before the 
last update. We´ll see... I will comment the result right here... in some 
hours I hope ;-)
```

Original comment by `rene.leh...@googlemail.com` on 23 Sep 2010 at 9:03

**Comment by GoogleCodeExporter** [comment]
```
Previous Version (1.3.3) still working  for me like a charm. Im on the HTC 
Legend with Android 2.1
```

Original comment by `gech...@gmail.com` on 23 Sep 2010 at 11:35

**Comment by GoogleCodeExporter** [comment]
```
I have the same problem with WebSMS v3.1.12 + O2 Connector V 1.3.6: 
Reinstalling both applications or only the connector gives a short release of 
the problem. After 2 or 3 SMS the error appears again.                               
```

Original comment by `tobias.z...@googlemail.com` on 23 Sep 2010 at 2:19

**Comment by GoogleCodeExporter** [comment]
```
I installed the "ConnectorO2-release.apk " patch now. Works fine after one SMS. 
We'll see :-)
```

Original comment by `tobias.z...@googlemail.com` on 23 Sep 2010 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
please try this build.

and please try to tell me the exact connector you're using while testing.
```

Original comment by `felix.bechstein` on 23 Sep 2010 at 5:45

Attachments:
- [ConnectorO2-release-2.apk](https://storage.googleapis.com/google-code-attachments/websmsdroid/issue-333/comment-22/ConnectorO2-release-2.apk)

**Comment by GoogleCodeExporter** [comment]
```
Unfortunately didn't the ConnectorO2-release.apk  not work. V1.3.3 works well.
How should the  "ConnectorO2-release-2.apk" be installed? Over Version 1.3.3? 
Or sould it be the newer version?
```

Original comment by `tobias.z...@googlemail.com` on 23 Sep 2010 at 7:22

**Comment by GoogleCodeExporter** [comment]
```
just install it. android will do it's work :)
```

Original comment by `felix.bechstein` on 23 Sep 2010 at 7:23

**Comment by GoogleCodeExporter** [comment]
```
Okay then. I tested now the connector 1.3.3 for several hours and several sms 
and it's still doing its job. now I'm gonna check the 
"ConnectorO2-release-2.apk". I will comment the result here again.

What about the update which is available in the market since last night? Which 
version is it and does it contain any fixes concerning this issue? Is testing 
the version mentioned above still necessary?

Note:
My device is HTC Desire with Android 2.1
```

Original comment by `rene.leh...@googlemail.com` on 24 Sep 2010 at 5:01

**Comment by GoogleCodeExporter** [comment]
```
the update (v1.3.6) in market is not working. and it is from mon, 20. sep.
please just test the last apk i attached here.
```

Original comment by `felix.bechstein` on 24 Sep 2010 at 5:06

**Comment by GoogleCodeExporter** [comment]
```
Hi there. The Update "ConnectorO2-release-2.apk " doesn't work. Same error.
```

Original comment by `tobias.z...@googlemail.com` on 24 Sep 2010 at 6:12

**Comment by GoogleCodeExporter** [comment]
```
Yup. I dont have any websms left, but i cant update the status to at least see 
the zero. Getting an java error
```

Original comment by `gech...@gmail.com` on 24 Sep 2010 at 7:31

**Comment by GoogleCodeExporter** [comment]
```
Wow, the "ConnectorO2-release-2.apk" works fine for me. (HTC Desire/Android 2.2)
I've sent two websms and made an almost incredible number of status-updates... 
;-)
```

Original comment by `kboss...@gmail.com` on 24 Sep 2010 at 8:09

**Comment by GoogleCodeExporter** [comment]
```
It´s me again ;-) after installing "ConnectorO2-release-2.apk" the first 
message I could send successfully. But as usually several hours later the same 
error occurs again "o2: Falscher Benutzernmane oder Passwort angegeben!" - both 
while updating the remaining free sms and while sendig a sms. (HTC Desire/ 
Android 2.1)

An interesting fact is that even if there is the error while status update it 
nevertheless shows me the actual number of remaining free sms after some 
seconds.
```

Original comment by `rene.leh...@googlemail.com` on 24 Sep 2010 at 8:49

**Comment by GoogleCodeExporter** [comment]
```
for those of you, who have a not working release2 installed, please send me 
your device's logs from websms' preferences.
```

Original comment by `felix.bechstein` on 24 Sep 2010 at 9:23

**Comment by GoogleCodeExporter** [comment]
```
I need to correct myself. After a few hours release 2 stopped working and 
throws the error again. I'll send the logs...
```

Original comment by `kboss...@gmail.com` on 25 Sep 2010 at 7:34

**Comment by GoogleCodeExporter** [comment]
```
ok. this seems to be a second issue.
the first one fixed is described here: issue #321

the second one should be something related to cookie handling.
i'll try to fix it this weekend.

please be patient and ready to test.
```

Original comment by `felix.bechstein` on 25 Sep 2010 at 8:10
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
this apk will not fix it, but print out some more debug information about your 
cookies etc.
please test it and send me the logs again, when it stops working.

btw: the connector is working here on my g1 as it should. never stopped 
working.. o.O
```

Original comment by `felix.bechstein` on 25 Sep 2010 at 8:47
- Changed state: **Started**

Attachments:
- [ConnectorO2-release-3.apk](https://storage.googleapis.com/google-code-attachments/websmsdroid/issue-333/comment-34/ConnectorO2-release-3.apk)

**Comment by GoogleCodeExporter** [comment]
```
after sending logs from release-3, please try to kill the process of the 
connector and try again.
```

Original comment by `felix.bechstein` on 25 Sep 2010 at 10:41

**Comment by GoogleCodeExporter** [comment]
```
please try the release-4.
i'm pretty sure, i found the problem:

the connector is trying to reuse the session as long as it does not get killed.
reinstalling the connector kills it. ;)

but if the session has turned invalid, the connector tries to reinitiate a new 
session. due to new cookie handling, it still has the old cookies stored and 
misses the successfull login.
i changed it to clear the internal cookie store when reinitating a session.

btw: it works on my g1, as the connector is killed by android after sending the 
message (due to low memory).

please test the connector and send logs if login fails again. (please add some 
lines to the logfile to let me know for what i've to look for)
```

Original comment by `felix.bechstein` on 25 Sep 2010 at 12:04
- Changed state: **ShouldBeFixed**

Attachments:
- [ConnectorO2-release-4.apk](https://storage.googleapis.com/google-code-attachments/websmsdroid/issue-333/comment-36/ConnectorO2-release-4.apk)

**Comment by GoogleCodeExporter** [comment]
```
Hi Felix, until now, your new fix seems to work fine. I could text several 
times and make some account enquiries without any mistake. 
```

Original comment by `tux.47...@googlemail.com` on 25 Sep 2010 at 4:49

**Comment by GoogleCodeExporter** [comment]
```
Update of the account balance just worked after installing the 4th release. 
Can't test sending sms, because my balance equals zero :>
```

Original comment by `gech...@gmail.com` on 25 Sep 2010 at 4:52

**Comment by GoogleCodeExporter** [comment]
```
ok, then leave it for a few ours and test, if it still works, when the session 
got invalid.
```

Original comment by `felix.bechstein` on 25 Sep 2010 at 5:00

**Comment by GoogleCodeExporter** [comment]
```
still OK...
```

Original comment by `tux.47...@googlemail.com` on 25 Sep 2010 at 6:41

**Comment by GoogleCodeExporter** [comment]
```
I guess, you finally tracked it down with release 4. I could send a couple of 
SMS with some hours inbetween and everything worked fine. Thanx a lot... ;-)
```

Original comment by `kboss...@gmail.com` on 25 Sep 2010 at 6:51

**Comment by GoogleCodeExporter** [comment]
```
Even the next day release 4 is still working. Obviously you finally fixed it. 
Thx for your good work.
```

Original comment by `rene.leh...@googlemail.com` on 26 Sep 2010 at 9:15

**Comment by GoogleCodeExporter** [comment]
```
just released it to market. thx for testing.
```

Original comment by `felix.bechstein` on 26 Sep 2010 at 11:37
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Just wanted to confirm that two users here have no problems any more after the 
latest market update. Apparently an effective fix.
```

Original comment by `hgm...@gmail.com` on 28 Sep 2010 at 7:41

**Comment by GoogleCodeExporter** [comment]
```
loaded today the actual version from the market and got the issue discribed on 
my android 1.6 phone... so this should be the version 4, like described above, 
shouldn't it? Version 1.3 doesn't work either...

```

Original comment by `Sturmsae...@gmail.com` on 18 Oct 2010 at 12:32

**Comment by GoogleCodeExporter** [comment]
```
Issue 363 has been merged into this issue.
```

Original comment by `felix.bechstein` on 18 Oct 2010 at 6:09

**Comment by GoogleCodeExporter** [comment]
```
Issue 362 has been merged into this issue.
```

Original comment by `felix.bechstein` on 18 Oct 2010 at 6:10

**Comment by GoogleCodeExporter** [comment]
```
are you sure, you entered your phone number in international format, like 
+4917912345678? you need to set the prefix too.

websms does automatically convert it to local number during login.
```

Original comment by `felix.bechstein` on 18 Oct 2010 at 6:11

**Comment by GoogleCodeExporter** [comment]
```
I'm absulutely sure, I entered both, the number in international format AND the 
+49 Prefix for Germany. I didn't know that your program would convert it 
backwards. I was just so helpless that I tried logging in on my PC. Then I 
happened to realize that o2 would not accept international format. That's why I 
tried to set the "username" aka "sender" to the regional format...
Maybe I'm just too stupid to handle the program righ,t for version 4 seems to 
please all the other people here who had this error... I also sent you my logs 
today.
I dearly hope,you can help me.
```

Original comment by `Sturmsae...@gmail.com` on 18 Oct 2010 at 8:37

**Comment by GoogleCodeExporter** [comment]
```
please uninstall the connector,
install it again from market. just to be sure all processes of it are killed.

then try it again.

if it does not work, please install the connector attached and send me your 
logs after failing.

ps: your logs mentioned above did not arrive.
```

Original comment by `felix.bechstein` on 19 Oct 2010 at 4:53
